pgrx/src/datum/array.rs

```diff
@@ -648,8 +660,16 @@ impl<'a, T: FromDatum> Iterator for ArrayTypedIterator<'a, T> {
             element
         }
     }
+
+    #[inline]
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let len = self.array.raw.len().saturating_sub(self.curr);
```

Comment:
Using saturating_sub is probably slightly paranoid here, our `next()` impls generally check based relative comparison to the length (not strict equality), so it seems more honest.